test/es-module/test-esm-loader-hooks.mjs

```diff
@@ -593,17 +593,19 @@ describe('Loader hooks', { concurrency: true }, () => {
       `
         import {MessageChannel} from 'node:worker_threads';
         import {register} from 'node:module';
+        import {setTimeout} from 'node:timers/promises';
         const {port1, port2} = new MessageChannel();
         port1.on('message', (msg) => {
           console.log('message', msg);
         });
         const result = register(
-          ${JSON.stringify(fixtures.fileURL('/es-module-loaders/hooks-initialize-port.mjs'))},
+          ${JSON.stringify(fixtures.fileURL('es-module-loaders/hooks-initialize-port.mjs'))},
           {data: port2, transferList: [port2]},
         );
         console.log('register', result);
 
         await import('node:os');
+        await setTimeout(99); // delay to limit flakiness
```

Comment:
isnt this just hiding a bug?

Reply:
Not really, race conditions are inherent to cross thread communication I think

Reply:
Wouldn't a user have to do this too? If so, this feels like a legit bug.

Reply:
We've known this since we started working on off-threading the loader hooks, it's documented here: https://github.com/nodejs/node/blob/6ad8318373664e5d9c4f4133c9393c59c720f6a9/doc/api/esm.md#L740-L743

Reply:
OH! Sorry, this is for a console log. Sorry, yes.

Reply:
Maybe `process._rawDebug` or `fs.writeFileSync(1`?